src/node_util.cc

```diff
@@ -146,6 +148,36 @@ void WatchdogHasPendingSigint(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void CreatePromise(const FunctionCallbackInfo<Value>& args) {
+  Local<Context> context = args.GetIsolate()->GetCurrentContext();
+  auto maybe_resolver = Promise::Resolver::New(context);
+  if (!maybe_resolver.IsEmpty())
+    args.GetReturnValue().Set(maybe_resolver.ToLocalChecked());
+}
+
+
+void PromiseResolve(const FunctionCallbackInfo<Value>& args) {
+  Local<Context> context = args.GetIsolate()->GetCurrentContext();
+  Local<Value> promise = args[0];
+  CHECK(promise->IsPromise());
+  if (promise.As<Promise>()->State() != Promise::kPending) return;
+  Local<Promise::Resolver> resolver = promise.As<Promise::Resolver>();  // sic
+  Maybe<bool> ret = resolver->Resolve(context, args[1]);
```

Comment:
Yea, but I’d say this doesn’t need to happen right now. It should be fine deferring that discussion until later (for example, it might be something we’d want to coordinate with node-chakracore etc.).